bchrpc/server.go

```diff
@@ -1248,10 +1324,25 @@ func (s *GrpcServer) SubscribeBlocks(req *pb.SubscribeBlocksRequest, stream pb.B
 			case *rpcEventBlockDisconnected:
 				// Search for all transactions.
 				block := event.(*rpcEventBlockDisconnected)
+				toSend := &pb.BlockNotification{}
+
+				// Serialize raw block using Bitcoin protocol encoding
+				if serializeBlock {
+					bytes, err := block.Block.Bytes()
+					if err != nil {
+						return status.Error(codes.Internal, "block serialization error")
+					}
 
-				toSend := &pb.BlockNotification{
-					Type:  pb.BlockNotification_DISCONNECTED,
-					Block: marshalBlockInfo(block.Block, 0, s.chainParams),
+					toSend.Type = pb.BlockNotification_CONNECTED
```

Comment:
NIT: this can be moved to line 1328 as it is shared by both the if and else.

Reply:
Oeps, even have a bug here. `toSend.Type = pb.BlockNotification_CONNECTED` should be  `pb.BlockNotification_DISCONNECTED`